From linusarver at gmail.com  Sun Oct 17 12:26:01 2010
From: linusarver at gmail.com (Linus Arver)
Date: Sun, 17 Oct 2010 12:26:01 -0700
Subject: [PATCH] Xext: panoramiXprocs: fix typo
Message-ID: <mailman.0.1287344880.11764.xorg-devel@lists.x.org>

This fixes a typo introduced in commit
80b5d3a3264d2c5167e5ac85a3b04af0f89cece1. The pointer pDst was changed
unintentionally to pWin from a copy/paste error. This resulted in all
QT-based apps and some tcl/tk ones (like fontforge) to crash X 1.9 on
starting up, when Xinerama was enabled.

Bug report: https://bbs.archlinux.org/viewtopic.php?id=106125

Signed-off-by: Elie Bleton <drozofil at gmail.com>
Tested-by: Linus Arver <linusarver at gmail.com>
---
 Xext/panoramiXprocs.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/Xext/panoramiXprocs.c b/Xext/panoramiXprocs.c
index 67b4030..d843168 100644
--- a/Xext/panoramiXprocs.c
+++ b/Xext/panoramiXprocs.c
@@ -634,7 +634,7 @@ int PanoramiXTranslateCoords(ClientPtr client)
     rep.dstX = x - pDst->drawable.x;
     rep.dstY = y - pDst->drawable.y;
     if((pDst == screenInfo.screens[0]->root) ||
-       (pWin->drawable.id == screenInfo.screens[0]->screensaver.wid))
+       (pDst->drawable.id == screenInfo.screens[0]->screensaver.wid))
     {
 	rep.dstX += screenInfo.screens[0]->x;
 	rep.dstY += screenInfo.screens[0]->y;
-- 
1.7.3.1
